src/test/java/seedu/tinner/storage/JsonAdaptedRoleTest.java

```diff
@@ -56,47 +57,47 @@ public void toModelType_nullRoleName_throwsIllegalValueException() {
     @Test
     public void toModelType_invalidStatus_throwsIllegalValueException() {
         JsonAdaptedRole role =
-                new JsonAdaptedRole(VALID_NAME, INVALID_STATUS, VALID_DEADLINE, VALID_DESCRIPTION, VALID_STIPEND);
+                new JsonAdaptedRole(VALID_NAME, INVALID_STATUS, VALID_REMINDER_DATE, VALID_DESCRIPTION, VALID_STIPEND);
         String expectedMessage = Status.MESSAGE_CONSTRAINTS;
         assertThrows(IllegalValueException.class, expectedMessage, role::toModelType);
     }
 
     @Test
     public void toModelType_nullStatus_throwsIllegalValueException() {
-        JsonAdaptedRole role = new JsonAdaptedRole(VALID_NAME, null, VALID_DEADLINE, VALID_DESCRIPTION,
+        JsonAdaptedRole role = new JsonAdaptedRole(VALID_NAME, null, VALID_REMINDER_DATE, VALID_DESCRIPTION,
                 VALID_STIPEND);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Status.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, role::toModelType);
     }
 
     @Test
-    public void toModelType_invalidDeadline_throwsIllegalValueException() {
+    public void toModelType_invalidReminderDate_throwsIllegalValueException() {
         JsonAdaptedRole role =
-                new JsonAdaptedRole(VALID_NAME, VALID_STATUS, INVALID_DEADLINE, VALID_DESCRIPTION, VALID_STIPEND);
-        String expectedMessage = Deadline.MESSAGE_CONSTRAINTS;
+                new JsonAdaptedRole(VALID_NAME, VALID_STATUS, INVALID_REMINDER_DATE, VALID_DESCRIPTION, VALID_STIPEND);
+        String expectedMessage = ReminderDate.MESSAGE_CONSTRAINTS;
         assertThrows(IllegalValueException.class, expectedMessage, role::toModelType);
     }
 
     @Test
-    public void toModelType_nullDeadline_throwsIllegalValueException() {
+    public void toModelType_nullReminderDate_throwsIllegalValueException() {
```

Comment:
Refer to previous comment on JsonAdaptedRole: this test case may not be required.